src/app/mesh/qgsmeshcalculatordialog.cpp

```diff
@@ -673,34 +673,36 @@ void QgsMeshCalculatorDialog::repopulateTimeCombos()
   const QgsMeshDataProvider *dp = layer->dataProvider();
 
   // extract all times from all datasets
-  QMap<QString, double> times;
+  QMap<qint64, double> times;
 
-  for ( int groupIndex = 0; groupIndex < dp->datasetGroupCount(); ++groupIndex )
+  const QList<int> groupIndexes = layer->datasetGroupsIndexes();
+  for ( int dsgi : groupIndexes )
   {
-    for ( int datasetIndex = 0; datasetIndex < dp->datasetCount( groupIndex ); ++datasetIndex )
+    int datasetCount = layer->datasetCount( QgsMeshDatasetIndex( dsgi, 0 ) );
+    for ( int datasetIndex = 0; datasetIndex < datasetCount; ++datasetIndex )
     {
-      const QgsMeshDatasetMetadata meta = dp->datasetMetadata( QgsMeshDatasetIndex( groupIndex, datasetIndex ) );
-      const double time = meta.time();
-      const QString timestr = layer->formatTime( time );
-
-      times[timestr] = time;
+      qint64 timeMs = layer->datasetRelativeTimeInMilliseconds( QgsMeshDatasetIndex( dsgi, datasetIndex ) );
+      if ( timeMs == INVALID_MESHLAYER_TIME )
+        continue;
+      const QgsMeshDatasetMetadata meta = dp->datasetMetadata( QgsMeshDatasetIndex( dsgi, datasetIndex ) );
+      times[timeMs] = meta.time();
     }
   }
 
-  // sort by text
-  auto keys = times.keys();
-  keys.sort();
+  const QList<qint64> keys = times.keys();
```

Comment:
It's no crucial for this (non performance oriented) code, but better practice here is to use an iterator directly on the map to avoid the cost of 
1. Iterating over the map to build a new list of key values
2. Iterating over the key list
3. Performing a map lookup for each of the key values